tests/misc/coroutines/src/issues/aidan/Issue27.hx

```diff
@@ -0,0 +1,57 @@
+package issues.aidan;
+
+import haxe.coro.context.Key;
+import haxe.coro.context.IElement;
+import haxe.coro.Coroutine;
+import haxe.coro.ICoroutineScope;
+
+class DebugName implements IElement<DebugName> {
+	static public var key:Key<DebugName> = Key.createNew("DebugName");
+
+	public var name:String;
+
+	public function new(name:String) {
+		this.name = name;
+	}
+
+	public function getKey() {
+		return key;
+	}
+
+	public function toString() {
+		return '[DebugName: $name]';
+	}
+}
+
+class Issue27 extends utest.Test {
+	function test() {
+		var log = [];
+
+		@:coroutine
+		function logDebug() {
+			Coroutine.suspend(cont -> {
+				log.push(cont.context.get(DebugName.key).name);
+				cont.resume(null, null);
+			});
+		}
+
+		@:coroutine
+		function modifyDebug(name:String) {
+			Coroutine.suspend(cont -> {
+				cont.context.get(DebugName.key).name = name;
+				cont.resume(null, null);
+			});
+		}
+		@:coroutine
+		function test(scope:ICoroutineScope) {
+			final coro = scope.with(new DebugName("first name")).start(_ -> {
+				logDebug();
+				modifyDebug("second name");
+				logDebug();
+				return log.join(", ");
+			});
+			return coro.await();
+		}
+		Assert.equals("first name, second name", Coroutine.runScoped(test));
```

Comment:
This made me realize that we don't have a way to set the initial context yet, i.e. the one that is set up by `BlockingCoroutine` at the moment. I'm not sure how to design this properly.

Reply:
We could add an extra argument to `runScoped` to allow specifying the initial context, we could also leave it as is and introduce some sort of `Coroutine.contex` function to ease adding context elements.

```haxe
Coroutine.runScoped(scope -> {
	Coroutine.context(some, extra, elements, scope -> {
		//
	});
});
```

Although, rest arguments have to be at the end, so we might want to change that signature a bit.